test/e2e/framework/util.go

```diff
@@ -1340,6 +1343,8 @@ func ExpectNoErrorWithRetries(fn func() error, maxRetries int, explain ...interf
 			return
 		}
 		e2elog.Logf("(Attempt %d of %d) Unexpected error occurred: %v", i+1, maxRetries, err)
+		// Dump the stacktrace while an error occurs
+		debug.PrintStack()
```

Comment:
If we're doing retries it seems like we expect errors and wouldn't want/need the stacktrace.

Can we move this down so that we only dump the trace here if we run out of retries and are actually bubbling up the error?

Reply:
My thought is that it may have different reasons to cause the retry fails each time. So for each retry I chose to print the stack.

Reply:
any comments?

Reply:
Although that _can_ happen; it has never been the case in my experience. Usually the same error happens 5 times so only the last one matters. OR you get an "error" which was somewhat expected (like polling for a service to come up) so you _see_ 4 errors (and now 4 entire stack traces) even though the test goes onto pass.

My preference would still be for just the trace at the end, but I don't feel _that_ strongly if you prefer this.

Reply:
This make more sense. I will move the printstack() out of the loop

Reply:
Let me know when you've made the changes and I'll re-review and remove the hold

Reply:
Ready to re-review